base/ubi8/dnf-wrapper.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+set -euo pipefail
+
+DNF_ORIGINAL_PATH=${DNF_ORIGINAL_PATH:-"/usr/bin/dnf.orig"}
+PODMAN_WRAPPER_PATH=${PODMAN_WRAPPER_PATH:-"/usr/bin/podman.wrapper"}
+PODMAN_ORIGINAL_PATH=${PODMAN_ORIGINAL_PATH:-"/usr/bin/podman.orig"}
+
+DNF_ARGS=( "$@" )
+
+TRUE=0
+FALSE=1
+
+exec_original_dnf() {
+    ${DNF_ORIGINAL_PATH} "${DNF_ARGS[@]}"
+}
+
+check_if_setup_needed() {
+    if command -v podman &> /dev/null && [ ! -e "$PODMAN_ORIGINAL_PATH" ]; then
+        return $TRUE
+    fi
+    return $FALSE
+}
+
+install_kubedock() {
+    curl -L https://github.com/joyrex2001/kubedock/releases/download/${KUBEDOCK_VERSION}/kubedock_${KUBEDOCK_VERSION}_linux_amd64.tar.gz | tar -C /usr/local/bin -xz --no-same-owner
```

Comment:
In the case of a custom image that extends the UBI, I've run into the following issue:

```
Podman is installed, installing Kubedock...
/bin/dnf: line 25: KUBEDOCK_VERSION: unbound variable

gzip: stdin: unexpected end of file
tar: Child returned status 1
tar: Error is not recoverable: exiting now
Error: building at STEP "RUN sudo dnf install -y podman": while running runtime: exit status 2
```

Setting `ENV KUBEDOCK_VERSION 0.15.5` in the Dockerfile before doing `sudo dnf install -y podman` doesn't resolve the issue either, unfortunately.

I'm not entirely sure yet why the KUBEDOCK_VERSION environment variable can't be accessed from `dnf-wrapper.sh`? It seems like  $KUBEDOCK_VERSION is set correctly, as the following Dockerfile works:

```Dockerfile
FROM <universal-base-image-from-PR>

USER 0

RUN echo $KUBEDOCK_VERSION

RUN curl -L https://github.com/joyrex2001/kubedock/releases/download/${KUBEDOCK_VERSION}/kubedock_${KUBEDOCK_VERSION}_linux_amd64.tar.gz
```

But the following does not:
```Dockerfile
FROM <universal-base-image-from-PR>

USER 0

RUN  sudo dnf install -y podman

USER 1000
```

Something odd is going on with the dnf-wrapper.sh script when trying to expand `${KUBEDOCK_VERSION}`

Reply:
It's working on the dogfooding cluster and my machine, could you share your base image?

Reply:
My base image is `quay.io/aobuchow/universal-developer-image:david-kubedock-pr-base` and here's the problematic Dockerfile:

```
FROM quay.io/aobuchow/universal-developer-image:david-kubedock-pr-base

USER 0

RUN echo $KUBEDOCK_VERSION

RUN sudo dnf install -y podman

USER 1000
```

I built the base image as well as the above Dockerfile with podman. Maybe there's some podman bug going on here. I'll try re-building with Docker using your instructions.

Reply:
I found my issue: I was invoking `dnf install -y podman` with `sudo` which was changing the bash environment. The following Dockerfile works as expected :grin: :

```Dockerfile
FROM quay.io/aobuchow/universal-developer-image:david-kubedock-pr-base

USER 0

RUN echo $KUBEDOCK_VERSION

RUN dnf install -y podman

USER 10001
```